src/Dapr.Actors/Communication/RemoteException.cs

```diff
@@ -147,8 +171,24 @@ private static bool TryDeserializeException(Stream data, out Exception result)
         {
             try
             {
-                result = (Exception)binaryFormatter.Deserialize(data);
-                return true;
+                using (var sr = new StreamReader(data, Encoding.UTF8))
+                {
+                    var str = sr.ReadToEnd();
+                    var dict = (Dictionary<string, object>)JsonSerializer.Deserialize(str, typeof(Dictionary<string, object>));
+                    var innerExceptionDict = (Dictionary<string, object>)JsonSerializer.Deserialize(dict[ExceptionInner].ToString(), typeof(Dictionary<string, object>));
+
+                    var innerExceptionType = Type.GetType(innerExceptionDict[ExceptionClassName].ToString());
+                    var innerException = (Exception)Activator.CreateInstance(innerExceptionType, innerExceptionDict[ExceptionMessage].ToString());
```

Comment:
This is the part of binary formatter that needs to be avoided. This allows a caller to send a type name over and HTTP request and we'll new up one of those. It's literally remote code execution.

What I expected to see was a design proposal for what the experience looks like when you subtract binary formatter. I don't think we should go down the path of trying to serialize arbitrary exception types. Normally (for ASP.NET Core) we would never send exception details like stack traces or inner exceptions over the wire.